pkg/kubernetes/pullsecrets.go

```diff
@@ -0,0 +1,78 @@
+package kubernetes
+
+import (
+	"context"
+
+	perrors "github.com/pkg/errors"
+)
+
+func (k *kubernetesDriver) EnsurePullSecret(
+	ctx context.Context,
+	pullSecretMetaData *PullSecretMetaData,
+	dockerConfigPath string,
+) error {
+	//todo: check if pull secret already exists, don't create it again
+	err := k.createPullSecret(ctx, *pullSecretMetaData, dockerConfigPath)
+	if err != nil {
+		return err
+	}
+
+	k.Log.Debugf("Pull secret '%s' created", pullSecretMetaData.pullSecretName)
+
+	err = k.addImagePullSecretsToServiceAccount(ctx, *pullSecretMetaData)
```

Comment:
Don't add the pull secret to the service account, instead set it directly on the pod